lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -123,17 +134,36 @@ def get(self, request, *args, **kwargs):
         monitoring_utils.set_custom_attribute('is_staff', request.user.is_staff)
         is_staff = bool(has_access(request.user, 'staff', course_key))
 
-        _, request.user = setup_masquerade(
-            request,
-            course_key,
-            staff_access=is_staff,
-            reset_masquerade_data=True
-        )
-
-        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        if student_id is None or student_id == request.user.id:
+            _, student = setup_masquerade(
+                request,
+                course_key,
+                staff_access=is_staff,
+                reset_masquerade_data=True
+            )
+        else:
+            # When a student_id is passed in, we display the progress page for the user
+            # with the provided user id, rather than the requesting user
+            try:
+                coach_access = has_ccx_coach_role(request.user, course_key)
+            except CCXLocatorValidationException:
+                coach_access = False
+
+            has_access_on_students_profiles = is_staff or coach_access
+            # Requesting access to a different student's profile
+            if not has_access_on_students_profiles:
+                raise Http404
+            try:
+                student = User.objects.get(id=student_id)
+            except User.DoesNotExist:
+                raise Http404
```

Comment:
pylint doesn't complain about a missing `from` here? Odd. But you can make this 404 slightly nicer by doing:

```
except User.DoesNotExist as exc:
    raise Http404 from exc
```